loadbalancer_interface/provides.py

```diff
@@ -43,21 +43,35 @@ def __init__(self, charm, relation_name):
         ):
             self.framework.observe(event, self._check_consumers)
 
+        self.state.set_default(follower_can_read_requests=False)
+
+    def follower_perms(self, *, read: bool = None) -> "LBConsumers":
+        """Set permissions on the relation for non-leader units"""
+        if read is not None:
+            self.state.follower_can_read_requests = read
+        return self
+
     def _check_consumers(self, event):
         if self.is_changed:
             self.on.requests_changed.emit()
 
     @cached_property
     def all_requests(self):
         """A list of all current consumer requests."""
-        if not self.unit.is_leader():
-            # Only the leader can process requests, so avoid mistakes
-            # by not even reading the requests if not the leader.
+        follower = not self.unit.is_leader()
+        if follower and not self.state.follower_can_read_requests:
+            """
+            It could be dangerous for the followers to respond
+            to requests, however, the followers may need access
+            to read the requested data.
+            * Only the leader should respond to the requests.
+            * Only the leader may read from relation.data[self.app]
+            """
             return []
         requests = []
         for relation in self.relations:
             schema = self._schema(relation)
-            local_data = relation.data[self.app]
+            local_data = {} if follower else relation.data[self.app]
```

Comment:
be aware, follower units cannot access the local application relation-data, only the unit data on a relation.  
This change hides `relation.data[self.app]`  from the follower by providing an empty dict insead